packages/actionmenu/src/react/index.js

```diff
@@ -193,104 +196,106 @@ const calcNestedMenuPosition = (menuWidth, origin) =>
     }
   }[origin])
 
-class Item extends React.Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      isNestedRendered: false
+function usePrevious(value) {
+  const ref = React.useRef()
+
+  React.useEffect(() => {
+    ref.current = value
```

Comment:
so yeah....weird huh. this is how i'm maintaining a reference to a prevProp value so that i can compare it later later with the new prop. the ref is used to store a mutable value across renders. 

here is the react docs on the `usePrevious` hook -> https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state

and docs on using refs for more that dom references -> https://reactjs.org/docs/hooks-faq.html#is-there-something-like-instance-variables

🤷‍♂ thoughts?